apps/extension/src/App/Accounts/ViewAccount/ViewAccount.tsx

```diff
@@ -1,99 +1,58 @@
 import {
-  Alert,
   GapPatterns,
   Heading,
   LinkButton,
-  Loading,
   Stack,
   ViewKeys,
 } from "@namada/components";
 import { AccountType, DerivedAccount } from "@namada/types";
-import { formatRouterPath } from "@namada/utils";
-import { AccountManagementRoute, TopLevelRoute } from "App/types";
-import { ParentAccount } from "background/keyring";
-import { useRequester } from "hooks/useRequester";
-import { QueryAccountsMsg } from "provider";
-import { useEffect, useState } from "react";
+import routes from "App/routes";
+import { AccountContext } from "context";
+import { useContext, useEffect, useState } from "react";
 import { useNavigate, useParams } from "react-router-dom";
-import { Ports } from "router";
 
 type ViewAccountUrlParams = {
   accountId: string;
-  type?: ParentAccount;
 };
 
 export const ViewAccount = (): JSX.Element => {
   const { accountId = "" } = useParams<ViewAccountUrlParams>();
-  const [loadingStatus, setLoadingStatus] = useState("");
-  const [accounts, setAccounts] = useState<DerivedAccount[]>([]);
+  const { accounts: accountStore } = useContext(AccountContext);
   const [parentAccount, setParentAccount] = useState<DerivedAccount>();
   const [transparentAddress, setTransparentAddress] = useState("");
   const [shieldedAddress, setShieldedAddress] = useState("");
-  const [error, setError] = useState("");
-
   const navigate = useNavigate();
-  const requester = useRequester();
-
-  const fetchAccounts = async (accountId: string): Promise<void> => {
-    setLoadingStatus("Loading...");
-    try {
-      const accounts = await requester.sendMessage(
-        Ports.Background,
-        new QueryAccountsMsg({ accountId })
-      );
-      setAccounts(accounts);
-
-      const parentAccount = accounts.find((account) => !account.parentId);
-      setParentAccount(parentAccount);
 
-      accounts.forEach((account) => {
-        if (
-          account.type === AccountType.Mnemonic ||
-          account.type === AccountType.Ledger
-        ) {
-          setTransparentAddress(account.address);
-        }
+  const searchAccounts = (accountId: string): void => {
```

Comment:
Nitpick: I think it might be a bit clearer  to return a value from this and maybe set state in the  effect :)